### ISSUE TITLE
Adding new waypoint into BlueSky Simulator

### ISSUE THREAD

**Issue by fadhliroslan** [ProfHoekstra/bluesky#177]
Hi, i found some missing waypoint and i want to add it in my simulator. How can i do that?

**Comment by ProfHoekstra** [comment]
You can make scenario file with a DEFWPT statement and call this at the beginning of your simulation.

**Comment by fadhliroslan** [comment]
It is possible if I want to add the waypoint without making a scenario file? I mean if I add the new waypoint in the existing waypoint file?

**Comment by ProfHoekstra** [comment]
Of course, but that has the risk it will be gone after an update.

What type of waypoint is it? If it is a VOR or a five character en-route waypoint, we can add it to the global database in the master version, just send us the info. But there is always the risk that it might be gone if we regenerate a new version.

Many local points i.e. lat/lons are not included in the waypoint file to keep the global file manageable and the database fast.

The normal way to add waypoints for a specific simulation is to make a dedicated scenario file for this. Check out for example the EHAM folder where we define a number of fixes for Amsterdam as well as a number of SIDs and STARs. When we simulate this area we simply call the DEFFIX.scn scenario from this folder from our scenario file (CALL DEFFIX) to define the custom waypoints. Then we can add the SIDs and STARs that use these to the route of an aircraft.

CALL SID06,KL204

Whether you edit a data file or a scenario file, the effect is in practice the same, except that when you keep the definition with your scenario files, you are independent and it will always work.

Best regards,
    Jacco Hoekstra


On 11 Dec 2018, at 07:30, fadhliroslan <notifications@github.com<mailto:notifications@github.com>> wrote:


It is possible if I want to add the waypoint without making a scenario file? I mean if I add the new waypoint in the existing waypoint file?

**Comment by fadhliroslan** [comment]
I had tried the suggestion you give but the phyton cant defined one item. It says that "global name 'bs' is not defined". What does it mean?

**Comment by ProfHoekstra** [comment]
What do you mean? Did you get this by using the DEFWPT command or did you edit source code?

Best regards/Met vriendelijke groeten,
    Jacco Hoekstra


On 11 Dec 2018, at 09:47, fadhliroslan <notifications@github.com<mailto:notifications@github.com>> wrote:


I had tried the suggestion you give but the phyton cant defined one item. It says that "global name 'bs' is not defined". What does it mean?

**Comment by fadhliroslan** [comment]
I used the DEFWPT command sir.

**Comment by jooste** [comment]
hi fadhliroslan,

I can use the DEFWPT command without problems.

Can you describe more exactly what you do, and can you give the complete text of the error that you get?

**Comment by jooste** [comment]
if you load this in a plain text file with either the `CALL`, `PCALL`, or the `IC` command, this should work. I assume that you don't try to load a .docx file in bluesky!

**Comment by fadhliroslan** [comment]
No, I did not. It just I cannot upload the other format here. Sorry sir. Can you teach me the CALL, PCALL, or the IC command? I did tried but maybe my command is wrong.

**Comment by jooste** [comment]
ok, good to know :) you could e.g. do:
   `CALL myfile.scn`
where myfile.scn, of course, is the file where you created with the DEFWPT commands.

**Comment by fadhliroslan** [comment]
Is it like this? I tried it but the simulator still cannot define the waypoint I made.
[NEW.SCN.txt](https://github.com/ProfHoekstra/bluesky/files/2667030/NEW.SCN.txt)

**Comment by jooste** [comment]
Ok, in a scenario file it would be `00:00:00.00>CALL NEW1`

**Comment by fadhliroslan** [comment]
The NEW1 is a coding that defines waypoints. Is it mean that I need to define it later on in scenario?

**Comment by jooste** [comment]
If NEW1 defines waypoints that you need for ADDWPT commands you obviously need to call it before those ADDWPT commands. So change the first line of your scenario to
`00:00:00>CALL NEW1`
This should work if you have a file called `NEW1.scn` in your `scenario` folder.

**Comment by ProfHoekstra** [comment]
A scenario file needs the extension .scn

Is that the problem?

Met vriendelijke groeten,
    Jacco Hoekstra


On 11 Dec 2018, at 11:39, fadhliroslan <notifications@github.com<mailto:notifications@github.com>> wrote:


The NEW1 is a coding that defines waypoints. Is it mean that I need to define it later on in scenario?

**Comment by fadhliroslan** [comment]
The extension wasn't the problem sir. According to Mr. Jooste, the command should be at the first. I tried it and compare it with the coding that you wrote which is the waypointExp.scn, and the outcome is the same. The picture below is the outcome using the waypointExp.scn coding you wrote. Maybe my coding was wrong, so I attached the file also.

[NEW.SCN.txt](https://github.com/ProfHoekstra/bluesky/files/2667202/NEW.SCN.txt)

 
![outcome](https://user-images.githubusercontent.com/45756731/49795915-36abd800-fd76-11e8-9112-400c3ace815a.png)
